apps/indexer/src/api/repositories/delegation-percentage.repository.ts

```diff
@@ -0,0 +1,87 @@
+import { db } from "ponder:api";
+import { daoMetricsDayBucket } from "ponder:schema";
+import { and, gte, lte, inArray, desc, asc } from "ponder";
+import { MetricTypesEnum } from "@/lib/constants";
+import type { SQL } from "drizzle-orm";
+
+export interface DaoMetricRow {
+  date: bigint;
+  daoId: string;
+  tokenId: string;
+  metricType: string;
+  high: bigint;
+}
+
+export interface DaoMetricsFilters {
+  startDate?: string;
+  endDate?: string;
+  orderDirection?: "asc" | "desc";
+  limit?: number;
+}
+
+export class DelegationPercentageRepository {
+  /**
+   * Fetches DELEGATED_SUPPLY and TOTAL_SUPPLY metrics from database
+   * @param filters - Date range and ordering filters
+   * @returns Array of metrics ordered by date
+   */
+  async getDaoMetricsByDateRange(
+    filters: DaoMetricsFilters,
+  ): Promise<DaoMetricRow[]> {
+    const { startDate, endDate, orderDirection = "asc", limit } = filters;
+
+    const queryFilters: (SQL | undefined)[] = [
+      inArray(daoMetricsDayBucket.metricType, [
+        MetricTypesEnum.DELEGATED_SUPPLY,
+        MetricTypesEnum.TOTAL_SUPPLY,
+      ]),
+    ];
+
+    if (startDate) {
+      queryFilters.push(gte(daoMetricsDayBucket.date, BigInt(startDate)));
+    }
+    if (endDate) {
+      queryFilters.push(lte(daoMetricsDayBucket.date, BigInt(endDate)));
+    }
+
+    const baseQuery = db
+      .select()
+      .from(daoMetricsDayBucket)
+      .where(and(...queryFilters.filter(Boolean)))
+      .orderBy(
+        orderDirection === "desc"
+          ? desc(daoMetricsDayBucket.date)
+          : asc(daoMetricsDayBucket.date),
+      );
+
+    const rows =
+      limit !== undefined ? await baseQuery.limit(limit) : await baseQuery;
+
+    return rows as DaoMetricRow[];
```

Comment:
it might be a good idea/ better separation of concerns if you return the database type natively from the repo function and map it to a custom interface later on, in the mapper layer

Reply:
try using [this drizzle way of fetching db entities](https://github.com/blockful/anticapture/blob/main/apps/indexer/src/api/repositories/transactions.repository.ts#L11) which is already typed